packages/flutter/lib/src/matrix_4_ext.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vector_math/vector_math_64.dart' show Matrix4;
+
+export 'package:vector_math/vector_math_64.dart' hide Colors;
+
+/// Extensions on [Matrix4] to easy migration to more efficient implementations.
+extension Matrix4Ext on Matrix4 {
+  /// Convenience wrapper over [translateByDouble].
+  void translateD(double x, [double y = 0, double z = 0]) {
+    translateByDouble(x, y, z, 1.0);
+  }
+
+  /// Convenience wrapper over [scaleByDouble].
+  void scaleD(double x, [double? y, double? z]) {
+    scaleByDouble(x, y ?? x, z ?? x, 1.0);
```

Comment:
one may expect that providing only an x value would only scale on the x axis.

I'd just drop the `by` to stay descriptive `scaleDouble` too

Reply:
This is mostly a brainstorm PR at the moment. Not sure if we want to spread these extensions all over... 🤷